eagle-xl/scripts/gas_masses.py

```diff
@@ -0,0 +1,108 @@
+"""
+Makes a histogram of the gas particle masses. Uses the swiftsimio library.
+"""
+
+import matplotlib.pyplot as plt
+import numpy as np
+
+from swiftsimio import load
+
+from unyt import unyt_quantity
+from matplotlib.colors import LogNorm
+from matplotlib.animation import FuncAnimation
+
+mass_bounds = [0.5, 8.5]
+
+
+def get_data(filename):
+    """
+    Grabs the data (masses in 10**6 Msun).
+    """
+
+    data = load(filename)
+
+    mass_gas = data.gas.masses.to("Msun") / 10 ** 6
+    mass_split = unyt_quantity(
+        1e4 * float(data.metadata.parameters["SPH:particle_splitting_mass_threshold"]),
```

Comment:
Same comment as above about the use of `get()`; also why is this 1e4 here?

Reply:
Because this is expressed in internal units. so 10^-4 * 10^10 = 10^6.

I could first convert the gas masses to 10^10 Msun and then multiply by 10^4 for converting to 10^6 Msun.

Reply:
Ah - can we then do:
```
mass_split = unyt_quantity(
    data.metadata.parameters.get("SPH:particle_splitting_mass_threshold", 0.0),
    units=data.units.mass,
).to("1e6 * Msun")
```

Reply:
Just trying to cover off a different set of mass units being chosen - however unlikely.